Mathlib/Algebra/Module/DirectLimitAndTensorProduct.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2023 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import Mathlib.Algebra.DirectLimit
+
+/-!
+# Tensor product and direct limits commute with each other.
+Given a family of `R`-modules `Gᵢ` with a family of compatible `R`-linear maps `fᵢⱼ : Gᵢ → Gⱼ` for
+every `i < j` and another `R`-module `M`, we have `(limᵢ Gᵢ) ⊗ M` and `lim (Gᵢ ⊗ M)` are isomorphic
+as `R`-modules.
+
+-/
+
+open TensorProduct Module.DirectLimit
+
+universe u v w
+
+variable {R : Type u} [CommRing R]
+variable {ι : Type v}
+variable [DecidableEq ι] [Preorder ι]
+variable {G : ι → Type w}
+
+namespace Module
+
+variable [∀ i, AddCommGroup (G i)] [∀ i, Module R (G i)]
+
+variable (f : ∀ i j, i ≤ j → G i →ₗ[R] G j)
+variable (M : Type w) [AddCommGroup M] [Module R M]
+
+/-- limᵢ (G i ⊗ M) -/
+local notation "Lim_G_tensor" =>
+  (DirectLimit (G · ⊗[R] M) <| fun i j h => LinearMap.rTensor M (f _ _ h))
+
+/-- lim G -/
+local notation "Lim_G" => (DirectLimit G f)
+
+/--
+the map `limᵢ (G i ⊗ M) → (limᵢ G i) ⊗ M` induced by the family of maps `G i ⊗ M → (limᵢ G i) ⊗ M`
+given by `gᵢ ⊗ m ↦ [gᵢ] ⊗ m`.
+-/
+noncomputable def directLimitOfTensorProductToTensorProductWithDirectLimit :
+    Lim_G_tensor →ₗ[R] Lim_G ⊗[R] M :=
+  lift _ _ _ _ (fun _ => (of _ _ _ _ _).rTensor M)
+    fun _ _ _ x => by refine' x.induction_on _ _ _ <;> aesop
+
+variable {M}
+@[simp] lemma directLimitOfTensorProductToTensorProductWithDirectLimit_apply_of_tmul
+    {i : ι} (g : G i) (m : M) :
+    directLimitOfTensorProductToTensorProductWithDirectLimit f M (of _ _ _ _ i (g ⊗ₜ m)) =
+    (of _ _ _ _ i g : Lim_G) ⊗ₜ m :=
+  lift_of (G := (G · ⊗[R] M)) _ _ (g ⊗ₜ m)
+
+variable (M)
+
+/--
+the map `(limᵢ G) ⊗ M → limᵢ (G i ⊗ M)` from the bilinear map `limᵢ Gᵢ → M → limᵢ (G i ⊗ M)` given
+by the family of maps `Gᵢ → M → limᵢ (G i ⊗ M)` where `gᵢ ↦ m ↦ [gᵢ ⊗ m]`
+
+-/
+noncomputable def tensorProductWithDirectLimitToDirectLimitOfTensorProduct :
+  Lim_G ⊗[R] M →ₗ[R] Lim_G_tensor :=
+TensorProduct.lift <| DirectLimit.lift _ _ _ _
+  (fun i =>
+    { toFun := (of R ι _ (fun i j h => (f _ _ h).rTensor M) i ∘ₗ TensorProduct.mk R _ _ .)
+      map_add' := by aesop
+      map_smul' := by aesop })
+  fun _ _ _ g => FunLike.ext _ _ (of_f (G := (G . ⊗[R] M)) (x := g ⊗ₜ .))
```

Comment:
@eric-wieser Do we have a policy whether to allow the regular dot `.` as function argument placeholders?

Reply:
I don't know if it's written anywhere, but it should certainly be a unicode .

Reply:
I didn't even realize that there are two different dots xd.